swift.en.md

```diff
@@ -0,0 +1,340 @@
+# Swift coding style
+
+## Introduction
+
+This document defines the conventions for writing Swift code at Cookpad Inc.
+
+This guide directed for Swift 3.0.2.
+
+## Objective
+
+This guides aim to accomplish these objectives.
+
+1. Clarify purposes 
+1. Improve readability
+1. Keep consistency
+1. Reduce side-effects
+
+## Code style
+
+- **[MUST]** Use four spaces for 1-level of indent.  Do not use the horizontal tab character.
+- **[MUST]** Add linebreak to end of files
+- **[MUST]** Place open brackets on same lines
+
+```swift
+// Bad
+func update() 
+{
+}
+
+// Good
+func update() {
+}
+```
+
+- **[MUST]** Use trailing comma on elements of collections
+    - Because of reducing file diffs
+
+```swift
+// Bad
+let ingredients = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half"
+]
+
+// Good
+let ingredients = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[MUST]** Do not use `()` wraps condition statements
+
+```swift
+// Bad
+if (a == 0) {
+}
+
+// Good
+if a == 0 {
+}
+```
+
+- **[MUST]** Do not use `;` on end of lines
+
+## Class, Struct
+
+- **[MUST]** Do not inherit `NSObject` except if needed to refer from Objective-C.
+- **[SHOULD]** Do not use `mutating` attribute for structs.
+    - Because structs should be immutable.
```

Comment:
Sometimes, having mutating attributes or function is kinda useful.

``` Swift
struct Rectangle {
    var topLeft: Point
    var size: Size
}

extension Rectangle {
    mutating translate(with offset: Point) {
        origin += offset
    }
    
    func translated(with offset: Point) -> Rectangle {
        var _self = self
        _self.translate(offset)
        return _self
    }
}

var myRect = Rectangle(topLeft: Point(0, 0), size: Size(100, 200))
myRect.translate(with: Point(10, 10))

let anotherRect = myRect.tanslated(with: Point(20, 20))
```

This 2 scenarios are all well adopted by Apple such as `Array` that has both `sort` and `sorted`.